xbmc/windowing/gbm/WinSystemGbm.h

```diff
@@ -60,11 +60,12 @@ class CWinSystemGbm : public CWinSystemBase
   virtual void Register(IDispResource *resource);
   virtual void Unregister(IDispResource *resource);
 
+  drm m_drm;
+
 protected:
-  CDRM m_DRM;
+  CDRMUtils *m_DRM;
```

Comment:
this can be `std::unique_ptr<CDRMUtils> m_DRM`, then in the class you do:

```c++
m_DRM.reset(new CDRMLegacy(&m_drm, &m_gbm));
if (!m_DRM->InitDrm())
{
  m_DRM.reset();
  ...
}
```

Reply:
Or even better, initialize m_DRM in the constructor to provide the assurance that it's never a nullptr

Reply:
Thanks @garbear I'll do what you say in your first comment because when atomic support is added we need to fallback to legacy if atomic fails.